lib/api/api_test.go

```diff
@@ -566,60 +566,354 @@ func TestHTTPLogin(t *testing.T) {
 		t.Fatal(err)
 	}
 	defer cancel()
+	url := baseURL + "/meta.js"
 
-	// Verify rejection when not using authorization
+	performRequest := func (username string, password string) *http.Response {
+		req, err := http.NewRequest("GET", url, nil)
+		if err != nil {
+			t.Fatal(err)
+		}
 
-	req, _ := http.NewRequest("GET", baseURL + "/meta.js", nil)
-	resp, err := http.DefaultClient.Do(req)
-	if err != nil {
-		t.Fatal(err)
+		if username != "" || password != "" {
+			req.SetBasicAuth(username, password)
+		}
+
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		return resp
 	}
+
+	assertHasSessionCookie := func (cookies []*http.Cookie) bool {
```

Comment:
nit: Just `hasSessionCookie` - I'd expect an `assert...` function to call `t.Fatal` (or not), not return a bool.

Reply:
Agreed, done!